File: backend/models/cv.py
"""CV data and response models."""
from typing import Optional, List
from pydantic import BaseModel, Field
from backend.models.personal import PersonalInfo
from backend.models.experience import Experience
from backend.models.education import Education, Skill


class CVData(BaseModel):
    """Complete CV data model."""

    personal_info: PersonalInfo
    experience: List[Experience] = []
    education: List[Education] = []
    skills: List[Skill] = []
    theme: Optional[str] = Field(
        default="classic",
        description="CV theme: accented, classic, colorful, creative, elegant, executive, minimal, modern, professional, or tech",
    )
    layout: Optional[str] = Field(
        default="classic-two-column",
        description="CV layout: classic-two-column, ats-single-column, modern-sidebar, section-cards-grid, career-timeline, project-case-studies, portfolio-spa, interactive-skills-matrix, academic-cv, dark-mode-tech",
    )
    target_company: Optional[str] = Field(
        default=None,
        max_length=200,
        description="Target company this CV was tailored for",
    )
    target_role: Optional[str] = Field(
        default=None,
        max_length=200,
        description="Target role this CV was tailored for",
    )


class CVResponse(BaseModel):
    """CV creation response."""

    cv_id: str
    filename: Optional[str] = None
    status: str = "success"


class CVListItem(BaseModel):
    """CV list item."""

    cv_id: str
    created_at: str
    updated_at: str
    person_name: Optional[str] = None
    filename: Optional[str] = None
    target_company: Optional[str] = None
    target_role: Optional[str] = None


class CVListResponse(BaseModel):
    """CV list response."""

    cvs: List[CVListItem]
    total: int

File: frontend/src/types/cv.ts
/** TypeScript types for CV data structure */

export interface Address {
  street?: string
  city?: string
  state?: string
  zip?: string
  country?: string
}

export interface PersonalInfo {
  name: string
  title?: string
  email?: string
  phone?: string
  address?: Address
  linkedin?: string
  github?: string
  website?: string
  summary?: string
  photo?: string
}

export interface Project {
  name: string
  description?: string
  highlights?: string[]
  technologies?: string[]
  url?: string
}

export interface Experience {
  title: string
  company: string
  start_date: string
  end_date?: string
  description?: string
  location?: string
  projects?: Project[]
}

export interface Education {
  degree: string
  institution: string
  year?: string
  field?: string
  gpa?: string
}

export interface Skill {
  name: string
  category?: string
  level?: string
}

export interface CVData {
  personal_info: PersonalInfo
  experience: Experience[]
  education: Education[]
  skills: Skill[]
  theme?:
    | 'accented'
    | 'classic'
    | 'colorful'
    | 'creative'
    | 'elegant'
    | 'executive'
    | 'minimal'
    | 'modern'
    | 'professional'
    | 'tech'
  layout?:
    | 'classic-two-column'
    | 'ats-single-column'
    | 'modern-sidebar'
    | 'section-cards-grid'
    | 'career-timeline'
    | 'project-case-studies'
    | 'portfolio-spa'
    | 'interactive-skills-matrix'
    | 'academic-cv'
    | 'dark-mode-tech'
  target_company?: string
  target_role?: string
}

export interface CVResponse {
  cv_id: string
  filename?: string
  status: string
}

export interface CVListItem {
  cv_id: string
  created_at: string
  updated_at: string
  person_name?: string
  filename?: string
  target_company?: string
  target_role?: string
}

export interface CVListResponse {
  cvs: CVListItem[]
  total: number
}

export interface ProfileData {
  personal_info: PersonalInfo
  experience: Experience[]
  education: Education[]
  skills: Skill[]
  updated_at?: string
}

export interface ProfileResponse {
  status: string
  message?: string
}

export interface ProfileListItem {
  name: string
  updated_at: string
}

export interface ProfileListResponse {
  profiles: ProfileListItem[]
}
